internal/data/talks.go

```diff
@@ -0,0 +1,58 @@
+package data
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"time"
+
+	"github.com/golang-insiders/site/internal/types"
+)
+
+type TalkService struct {
+	DB *sql.DB
+}
+
+func (tm *TalkService) Insert(t *types.Talk) error {
+	query := `
+		INSERT INTO talks (twitter_username, title, summary, timezone)
+		VALUES ($1, $2, $3, $4)
+		RETURNING id, created_at`
+
+	args := []interface{}{t.TwitterUsername, t.Title, t.Summary, t.Timezone}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+
+	return tm.DB.QueryRowContext(ctx, query, args...).Scan(&t.ID, &t.CreatedAt)
+}
+
+func (tm *TalkService) GetByID(id int) (*types.Talk, error) {
```

Comment:
We should put context as the first argument?
```
func (tm *TalkService) GetByID(ctx context.Context, id int) (*types.Talk, error)
```

Also maybe we need to extract GetByID, Insert, etc, and other methods into the interface? And implement it for `TalkService`? To make the tests for those methods, for example?